file-formats/smtg/type/zob_aff_smtg_v1.intf.abap

```diff
@@ -0,0 +1,156 @@
+interface zob_aff_smtg_v1
+  public.
+
+  " Email Template Metadata --------------------------------------------------------------------------------------------
+  types:
+    "! <p class="shorttext">Administrative Data</p>
+    "! Contains administrative details such as creation and last change information.
+    begin of ty_administrative_data,
+      "! <p class="shorttext">Created By</p>
+      "! Created by
+      "! $showAlways
+      created_by      type string,
+      "! <p class="shorttext">Last Changed By</p>
+      "! last changed by
+      "! $showAlways
+      last_changed_by type string,
+    end of ty_administrative_data.
+
+  types:
+    "! <p class="shorttext">Header</p>
+    "! Contains identifying and descriptive information about the template.
+    begin of ty_template_header,
+      "! <p class="shorttext">Template Name</p>
+      "! Unique identifier for the template
+      "! $required
+      template_id  type smtg_tmpl_name,
```

Comment:
smtg_tmpl_name can not be used in AFF. You can only use types which are mentioned [here](https://github.com/SAP/abap-file-formats/blob/main/docs/json.md#writing-json-schema-with-abap-types). 
"The ABAP types are self-contained, so it is possible to work on them in any system (e.g., in an SAP BTP, ABAP environment system)."

There are some more types which need to be changed.